### ISSUE TITLE
Required changes for passing parameters when communicating with Soroban RPC.

### ISSUE THREAD

**Issue by Shaptic** [StellarCN/py-stellar-base#882]
**TL;DR**: Array-based passing will no longer be allowed in Soroban RPC.

## Soroban RPC Changes

With the introduction of [stellar/soroban-rpc#13](https://github.com/stellar/soroban-rpc/issues/13), the Soroban RPC server changes the way it interprets the [JSON-RPC spec](https://www.jsonrpc.org/specification). 

Specifically, in [_Section 4.2_](https://www.jsonrpc.org/specification#parameter_structures), it states that parameters can be passed _by position_ (i.e. through an array) and _by name_ (i.e. through an object). However, endpoints that support optional parameters will not work with position-(array-)based calls. Thus, we must move exclusively to object-based calls to support optional parameters in the future. 

We saw this issue introduced in Soroban RPC [v20.1.0](https://github.com/stellar/soroban-tools/releases/tag/v20.1.0) as part of [soroban-tools#1131](https://github.com/stellar/soroban-tools/pull/1131): adding an optional parameter should be a non-breaking change, but because of array-based parameter parsing, this resulted in a breaking change in any environment using that method.

### Expectations

You should remove all instances of array-based parameter passing in your low-level JSON-RPC code.

For example, RPC calls should change like this:

```diff
 {
   jsonrpc: "2.0",
   id: 1,
   method: "getTransaction",
-  params: [ "d1344aacdb36949b0164b9b97a8a4961bd9ccadcf2c3b562c93b09bc8651dad7" ]
+  params: { hash: "d1344aacdb36949b0164b9b97a8a4961bd9ccadcf2c3b562c93b09bc8651dad7" }
 }
```

### References

 * The [RPC documentation](https://soroban.stellar.org/api/methods) outlines all of the named parameters necessary for object-based parameters.
 * An example implementation in JavaScript: [stellar-sdk#924](https://github.com/stellar/js-stellar-sdk/pull/924).

**Comment by overcat** [comment]
The Python SDK has already followed the relevant specifications, so no modifications are needed.
